src/math/TransformBoundedField.cc

```diff
@@ -0,0 +1,172 @@
+                                                         CatalogVector const& catalogs) const {
+        LSST_ARCHIVE_ASSERT(catalogs.size() == 1u);
+        LSST_ARCHIVE_ASSERT(catalogs.front().size() == 1u);
+        table::BaseRecord const& record = catalogs.front().front();
+        PersistenceHelper const keys(record.getSchema());
+        // NOTE: needed invert=false in case min=-1, max=0 (empty bbox). See RFC-324 and DM-10200
+        geom::Box2I bbox(record.get(keys.bboxMin), record.get(keys.bboxMax), false);
+        auto frameSetStr = record.get(keys.frameSet);
+        auto stream = ast::StringStream(frameSetStr);
+        auto astObjectPtr = ast::Channel(stream).read();
+        auto frameSet = std::dynamic_pointer_cast<ast::FrameSet>(astObjectPtr);
+        if (!frameSet) {
+            throw LSST_EXCEPT(pex::exceptions::RuntimeError,
+                              "The contents of the frameSet field could not be parsed as an ast::FrameSet");
+        }
+        auto transform = geom::Transform<geom::Point2Endpoint, geom::GenericEndpoint>(*frameSet);
+        return std::make_shared<TransformBoundedField>(bbox, transform);
+    }
+
+    TransformBoundedFieldFactory(std::string const& name) : afw::table::io::PersistableFactory(name) {}
+};
+
+std::string getTransformBoundedFieldPersistenceName() { return "TransformBoundedField"; }
+
+TransformBoundedFieldFactory registration(getTransformBoundedFieldPersistenceName());
+
+}  // namespace
+
+std::string TransformBoundedField::getPersistenceName() const {
+    return getTransformBoundedFieldPersistenceName();
+}
+
+std::string TransformBoundedField::getPythonModule() const { return "lsst.afw.math"; }
+
+void TransformBoundedField::write(OutputArchiveHandle& handle) const {
+    PersistenceHelper const keys;
+    table::BaseCatalog catalog = handle.makeCatalog(keys.schema);
+    std::shared_ptr<table::BaseRecord> record = catalog.addNew();
+    record->set(keys.bboxMin, getBBox().getMin());
+    record->set(keys.bboxMax, getBBox().getMax());
+    record->set(keys.frameSet, getTransform().getFrameSet()->show());
+    handle.saveCatalog(catalog);
+}
+
+PTR(BoundedField) TransformBoundedField::operator*(double const scale) const {
+    auto zoomMap = ast::ZoomMap(1, scale);
+    auto newMapping = getTransform().getFrameSet()->then(zoomMap);
+    auto newTransform = Transform(newMapping);
+    return std::make_shared<TransformBoundedField>(getBBox(), newTransform);
+}
+
+bool TransformBoundedField::operator==(BoundedField const& rhs) const {
+    auto rhsCasted = dynamic_cast<TransformBoundedField const*>(&rhs);
+    if (!rhsCasted) return false;
+
+    return getBBox() == rhsCasted->getBBox() &&
+           *(getTransform().getFrameSet()) == *(rhsCasted->getTransform().getFrameSet());
+}
+
+std::string TransformBoundedField::toString() const {
+    std::ostringstream os;
+    os << "TransformBoundedField (containing " << _transform.getFrameSet()->getNFrame() << " frames)";
```

Comment:
The internal string representation is many lines of text -- not at all suitable for a basic representation. Unfortunately I don't know of a good way to summarize a frame set. @parejkoj requested the # of frames as better than nothing, but I disagree; I think it is meaningless (and will usually be 2). I'm going to drop it for now.